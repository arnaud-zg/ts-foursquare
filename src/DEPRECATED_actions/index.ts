import { TLifeAction } from './life'

export type TRootAction = TLifeAction

export * from './life'
