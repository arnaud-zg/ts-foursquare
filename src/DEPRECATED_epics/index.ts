export * from './life'
export * from './lists'
export * from './photos'
export * from './venues'
