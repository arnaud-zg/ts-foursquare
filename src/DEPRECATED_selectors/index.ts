export * from './life'
export * from './photos'
export * from './request'
export * from './status'
export * from './venues'
