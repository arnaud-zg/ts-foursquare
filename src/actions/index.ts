import { TLifeAction } from './life'
import { TVenuesAction } from './venues'

export type TRootActionType = TVenuesAction | TLifeAction

export type TRootAction = TVenuesAction | TLifeAction

export * from './life'
export * from './venues'
