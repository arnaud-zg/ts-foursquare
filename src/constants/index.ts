export const PROJECT_NAME = 'ts-foursquare'
