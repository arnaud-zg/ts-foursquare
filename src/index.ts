require('isomorphic-fetch')

export * from './standalone'
export * from './utils'

export * from './DEPRECATED_actions'
export * from './DEPRECATED_reducers'
export * from './DEPRECATED_selectors'
