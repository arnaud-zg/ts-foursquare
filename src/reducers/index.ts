import { NVenues } from '../../types/venues'

export type TRootState = {
  venues: NVenues.IVenuesState
}
