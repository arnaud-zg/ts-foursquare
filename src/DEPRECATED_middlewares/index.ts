import { combineEpics, createEpicMiddleware } from 'redux-observable'
import { NStore } from '../../types'
import { TRootAction } from '../DEPRECATED_actions'
import {
  getAccessTokenEpic,
  getPhotosDetailsEpic,
  getVenuesCategoriesEpic,
  getVenuesExploreEpic,
  getVenuesLikesEpic,
  getVenuesListedEpic,
  getVenuesNextVenuesEpic,
  getVenuesSearchEpic,
  getVenuesSimilarEpic,
  getVenuesSuggestCompletionEpic,
  getVenuesTrendingEpic,
} from '../DEPRECATED_epics'

export const rootEpic = combineEpics(
  getAccessTokenEpic,
  getPhotosDetailsEpic,
  getVenuesCategoriesEpic,
  getVenuesExploreEpic,
  getVenuesLikesEpic,
  getVenuesListedEpic,
  getVenuesNextVenuesEpic,
  getVenuesSearchEpic,
  getVenuesSimilarEpic,
  getVenuesSuggestCompletionEpic,
  getVenuesTrendingEpic
)
export const epicMiddleware = createEpicMiddleware<
  TRootAction,
  TRootAction,
  NStore.IState
>()
