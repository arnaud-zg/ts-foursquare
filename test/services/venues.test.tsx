import { StateObservable } from 'redux-observable'
import { Subject } from 'rxjs/internal/Subject'
import {
  getVenuesExploreAsync,
  getVenuesLikesAsync,
  getVenuesListedAsync,
  getVenuesNextVenuesAsync,
  getVenuesSearchAsync,
  getVenuesSimilarAsync,
  getVenuesSuggestCompletionAsync,
  getVenuesTrendingAsync,
} from '../../src/actions'
import { initialState } from '../../src/reducers'
import {
  getObservableVenuesCategories,
  getObservableVenuesExplore,
  getObservableVenuesLikes,
  getObservableVenuesListed,
  getObservableVenuesNextVenues,
  getObservableVenuesSearch,
  getObservableVenuesSimilar,
  getObservableVenuesSuggestCompletion,
  getObservableVenuesTrending,
} from '../../src/services'
import { NStore } from '../../types'

describe('services/getObservableVenuesCategories', () => {
  it('should get an observable instance', () => {
    const state$: StateObservable<NStore.IState> = new StateObservable(
      new Subject(),
      initialState
    )

    expect(getObservableVenuesCategories({ state$ })).toMatchSnapshot()
  })
})

describe('services/getObservableVenuesExplore', () => {
  it('should get an observable instance', () => {
    const action = getVenuesExploreAsync.request({ ll: '40.7099,-73.9622' })
    const state$: StateObservable<NStore.IState> = new StateObservable(
      new Subject(),
      initialState
    )

    expect(getObservableVenuesExplore({ action, state$ })).toMatchSnapshot()
  })
})

describe('services/getObservableVenuesLikes', () => {
  it('should get an observable instance', () => {
    const action = getVenuesLikesAsync.request({
      venueId: '3fd66200f964a5209beb1ee3',
    })
    const state$: StateObservable<NStore.IState> = new StateObservable(
      new Subject(),
      initialState
    )

    expect(getObservableVenuesLikes({ action, state$ })).toMatchSnapshot()
  })
})

describe('services/getObservableVenuesListed', () => {
  it('should get an observable instance', () => {
    const action = getVenuesListedAsync.request({
      venueId: '3fd66200f964a5209beb1ee3',
    })
    const state$: StateObservable<NStore.IState> = new StateObservable(
      new Subject(),
      initialState
    )

    expect(getObservableVenuesListed({ action, state$ })).toMatchSnapshot()
  })
})

describe('services/getObservableVenuesNextVenues', () => {
  it('should get an observable instance', () => {
    const action = getVenuesNextVenuesAsync.request({
      venueId: '3fd66200f964a5209beb1ee3',
    })
    const state$: StateObservable<NStore.IState> = new StateObservable(
      new Subject(),
      initialState
    )

    expect(getObservableVenuesNextVenues({ action, state$ })).toMatchSnapshot()
  })
})

describe('services/getObservableVenuesSearch', () => {
  it('should get an observable instance', () => {
    const action = getVenuesSearchAsync.request({ ll: '40.7099,-73.9622' })
    const state$: StateObservable<NStore.IState> = new StateObservable(
      new Subject(),
      initialState
    )

    expect(getObservableVenuesSearch({ action, state$ })).toMatchSnapshot()
  })
})

describe('services/getObservableVenuesSimilar', () => {
  it('should get an observable instance', () => {
    const action = getVenuesSimilarAsync.request({
      venueId: '3fd66200f964a5209beb1ee3',
    })
    const state$: StateObservable<NStore.IState> = new StateObservable(
      new Subject(),
      initialState
    )

    expect(getObservableVenuesSimilar({ action, state$ })).toMatchSnapshot()
  })
})

describe('services/getObservableVenuesSuggestCompletion', () => {
  it('should get an observable instance', () => {
    const action = getVenuesSuggestCompletionAsync.request({
      ll: '40.7099,-73.9622',
      query: 'Burger',
    })
    const state$: StateObservable<NStore.IState> = new StateObservable(
      new Subject(),
      initialState
    )

    expect(
      getObservableVenuesSuggestCompletion({ action, state$ })
    ).toMatchSnapshot()
  })
})

describe('services/getObservableVenuesTrending', () => {
  it('should get an observable instance', () => {
    const action = getVenuesTrendingAsync.request({ ll: '40.7099,-73.9622' })
    const state$: StateObservable<NStore.IState> = new StateObservable(
      new Subject(),
      initialState
    )

    expect(getObservableVenuesTrending({ action, state$ })).toMatchSnapshot()
  })
})
