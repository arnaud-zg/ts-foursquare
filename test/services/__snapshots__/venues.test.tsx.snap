// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`services/getObservableVenuesCategories should get an observable instance 1`] = `
Observable {
  "_isScalar": false,
  "operator": CatchOperator {
    "caught": [Circular],
    "selector": [Function],
  },
  "source": Observable {
    "_isScalar": false,
    "operator": SwitchMapOperator {
      "project": [Function],
    },
    "source": Observable {
      "_isScalar": false,
      "_subscribe": [Function],
    },
  },
}
`;

exports[`services/getObservableVenuesExplore should get an observable instance 1`] = `
Observable {
  "_isScalar": false,
  "operator": CatchOperator {
    "caught": [Circular],
    "selector": [Function],
  },
  "source": Observable {
    "_isScalar": false,
    "operator": SwitchMapOperator {
      "project": [Function],
    },
    "source": Observable {
      "_isScalar": false,
      "_subscribe": [Function],
    },
  },
}
`;

exports[`services/getObservableVenuesLikes should get an observable instance 1`] = `
Observable {
  "_isScalar": false,
  "operator": CatchOperator {
    "caught": [Circular],
    "selector": [Function],
  },
  "source": Observable {
    "_isScalar": false,
    "operator": SwitchMapOperator {
      "project": [Function],
    },
    "source": Observable {
      "_isScalar": false,
      "_subscribe": [Function],
    },
  },
}
`;

exports[`services/getObservableVenuesListed should get an observable instance 1`] = `
Observable {
  "_isScalar": false,
  "operator": CatchOperator {
    "caught": [Circular],
    "selector": [Function],
  },
  "source": Observable {
    "_isScalar": false,
    "operator": SwitchMapOperator {
      "project": [Function],
    },
    "source": Observable {
      "_isScalar": false,
      "_subscribe": [Function],
    },
  },
}
`;

exports[`services/getObservableVenuesNextVenues should get an observable instance 1`] = `
Observable {
  "_isScalar": false,
  "operator": CatchOperator {
    "caught": [Circular],
    "selector": [Function],
  },
  "source": Observable {
    "_isScalar": false,
    "operator": SwitchMapOperator {
      "project": [Function],
    },
    "source": Observable {
      "_isScalar": false,
      "_subscribe": [Function],
    },
  },
}
`;

exports[`services/getObservableVenuesSearch should get an observable instance 1`] = `
Observable {
  "_isScalar": false,
  "operator": CatchOperator {
    "caught": [Circular],
    "selector": [Function],
  },
  "source": Observable {
    "_isScalar": false,
    "operator": SwitchMapOperator {
      "project": [Function],
    },
    "source": Observable {
      "_isScalar": false,
      "_subscribe": [Function],
    },
  },
}
`;

exports[`services/getObservableVenuesSimilar should get an observable instance 1`] = `
Observable {
  "_isScalar": false,
  "operator": CatchOperator {
    "caught": [Circular],
    "selector": [Function],
  },
  "source": Observable {
    "_isScalar": false,
    "operator": SwitchMapOperator {
      "project": [Function],
    },
    "source": Observable {
      "_isScalar": false,
      "_subscribe": [Function],
    },
  },
}
`;

exports[`services/getObservableVenuesSuggestCompletion should get an observable instance 1`] = `
Observable {
  "_isScalar": false,
  "operator": CatchOperator {
    "caught": [Circular],
    "selector": [Function],
  },
  "source": Observable {
    "_isScalar": false,
    "operator": SwitchMapOperator {
      "project": [Function],
    },
    "source": Observable {
      "_isScalar": false,
      "_subscribe": [Function],
    },
  },
}
`;

exports[`services/getObservableVenuesTrending should get an observable instance 1`] = `
Observable {
  "_isScalar": false,
  "operator": CatchOperator {
    "caught": [Circular],
    "selector": [Function],
  },
  "source": Observable {
    "_isScalar": false,
    "operator": SwitchMapOperator {
      "project": [Function],
    },
    "source": Observable {
      "_isScalar": false,
      "_subscribe": [Function],
    },
  },
}
`;
